internal/gqlclient/gqlclient.go

```diff
@@ -304,20 +305,46 @@ func (c *Client) runWithFiles(ctx context.Context, gqlReq RequestWithFiles, resp
 	}
 	req = req.WithContext(ctx)
 	c.Log(fmt.Sprintf(">> Raw Request: %s\n", req.URL.String()))
-	res, resErr := http.DefaultClient.Do(req)
-	if reqErr := <-reqErrChan; reqErr != nil {
-		return reqErr
+
+	var res *http.Response
+	resErrChan := make(chan error)
+	go func() {
+		var err error
+		res, err = http.DefaultClient.Do(req)
+		resErrChan <- err
+	}()
+
+	// Due to the streaming uploads the request error can happen both before and after the http request itself, hence
+	// the creative select case you see before you.
```

Comment:
Without this the request would hang forever if there are input issues, as it would hang waiting for a response that would never come.